assets/js/base/context/hooks/use-store-events.ts

```diff
@@ -39,11 +44,13 @@ export const useStoreEvents = (): {
 	const dispatchCheckoutEvent = useCallback(
 		( eventName, eventParams = {} ) => {
 			try {
+				const cartParam = currentStoreCart.current;
+
 				doAction(
 					`experimental__woocommerce_blocks-checkout-${ eventName }`,
 					{
 						...eventParams,
-						storeCart,
+						cartParam,
 					}
```

Comment:
I know why I did this (it's used as a dependency in various places and I didn't want to trigger the effects when the cart changed). I think to avoid major changes, I'll just store the cart as a ref so the callback function can use the ref instead. Adding tests I'm still not confident with and I know it will take me the rest of the week to fathom.

I did a quick test to make sure I understand refs and callbacks so it will have the latest cart object available (https://codesandbox.io/s/clever-brook-i8fuy?file=/src/app.js).

Reply:
I've restored the `storeCart` parameter name here; that was an oversight on my part. It's still using the ref.

Reply:
> I know tests would be good, but I'm still not confident writing tests from scratch and I know it will take me the rest of the week to fathom.

I'm okay with it not blocking the PR, but I do think it'd be worthwhile doing this as a followup. We need more confidence when changing things and automated tests are a good way to get that.